bbid.py

```diff
@@ -36,12 +39,15 @@ def download(url,output_dir,retry=False):
 		md5 = hashlib.md5()
 		md5.update(image)
 		md5_key = md5.hexdigest()
+		img_hash_lock.acquire()
 		if md5_key in image_md5s:
```

Comment:
Is this the actual problem you experienced? Chance of this happening seems to be VERY small, since only two very simple  operations (if + saving filename) are happening between thread acquiring and releasing